app/controllers/user_sessions_controller.rb

```diff
@@ -66,75 +66,82 @@ def create
         end
       end
     else
-      params[:user_session][:username] = params[:openid] if params[:openid] # second runthrough must preserve username
       username = params[:user_session][:username] if params[:user_session]
-      @user = User.find_by(username: username)
-
-      # try finding by email, if that exists
-      if @user.nil? && !User.where(email: username).empty?
-        @user = User.find_by(email: username)
-        params[:user_session][:username] = @user.username
-      end
-
-      if @user.nil?
-        flash[:warning] = "There is nobody in our system by that name, are you sure you have the right username?"
-        redirect_to '/login'
-      elsif params[:user_session].nil? || @user&.drupal_user&.status == 1
-        # an existing Rails user
-        if params[:user_session].nil? || @user
-          if @user&.crypted_password.nil? # the user has not created a pwd in the new site
-            params[:user_session][:openid_identifier] = 'https://old.publiclab.org/people/' + username + '/identity' if username
-            params[:user_session].delete(:password)
-            params[:user_session].delete(:username)
-            params[:openid] = username # pack up username for second runthrough
-          end
-          @user_session = UserSession.new(username: params[:user_session][:username],
-                                          password: params[:user_session][:password],
-                                          remember_me: params[:user_session][:remember_me])
-          saved = @user_session.save do |result|
-            if result
-              # replace this with temporarily saving pwd in session,
-              # and automatically saving it in the user record after login is completed
-              if current_user.crypted_password.nil? # the user has not created a pwd in the new site
-                flash[:warning] = I18n.t('user_sessions_controller.create_password_for_new_site')
-                redirect_to '/profile/edit'
-              else
-                flash[:notice] = I18n.t('user_sessions_controller.logged_in')
-                if session[:openid_return_to] # for openid login, redirects back to openid auth process
-                  return_to = session[:openid_return_to]
-                  session[:openid_return_to] = nil
-                  redirect_to return_to
-                elsif session[:return_to]
-                  return_to = session[:return_to]
-                  session[:return_to] = nil
-                  redirect_to return_to
-                elsif params[:return_to]
-                  redirect_to params[:return_to]
+      u = User.find_by(username: username)
+      if u && u.password_checker != 0
+        n = u.password_checker
+        hash = { 1 => "Facebook", 2 => "Github", 3 => "Google", 4 => "Twitter"  }
```

Comment:
Is it possible to use the style `1: "Facebook"` here?

Reply:
Pushing the changes.
Thanks

Reply:
I can't make integers as symbols as at another place I am using the reverse hash. In reverse hash I am mapping providers with integer values.

Reply:
ok! thanks for trying!